### PR TITLE
[Improve] support gradcam

### PR DESCRIPTION
**Pull request by cir7**
## Motivation
[Improvement] support gradcam

### CODE REVIEW

**Review by Dai-Wenxun — APPROVED** [review]

File: mmaction/apis/inference.py [hunk 1]
```diff
@@ -20,11 +20,11 @@ def init_recognizer(config: Union[str, Path, mmengine.Config],
     """Initialize a recognizer from config file.
 
     Args:
-        config (Union[str, :obj:`Path`, :obj:`mmengine.Config`]): Config file
+        config (str or :obj:`Path` or :obj:`mmengine.Config`): Config file
             path, :obj:`Path` or the config object.
         checkpoint (str, optional): Checkpoint path/url. If set to None,
             the model will not load any weights. Defaults to None.
-        device (Union[str, torch.device]): The desired device of returned
+        device (str or torch.device): The desired device of returned
```
Comment (ly015) [inline mmaction/apis/inference.py:27]:
```suggestion
        device (str | torch.device): The desired device of returned
```

File: mmaction/utils/gradcam_utils.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # Copyright (c) OpenMMLab. All rights reserved.
 import torch
+import torch.nn as nn
```
Comment (ly015) [inline mmaction/utils/gradcam_utils.py:3]: What about moving this to 'visualization/grad_cam.py'?

File: tests/models/base.py [hunk 3]
```diff
@@ -104,31 +104,6 @@ def random_label(n):
     return dict(img=[img], proposals=[proposals], img_metas=[img_metas])
 
 
-def generate_gradcam_inputs(input_shape=(1, 3, 3, 224, 224), model_type='2D'):
```
Comment (ly015) [inline tests/models/base.py:107]: This file should be move to 'mmaction/testing/'

File: mmaction/utils/gradcam_utils.py (hunk 2 above)
Comment (cir7) [reply mmaction/utils/gradcam_utils.py:3]: I will reorganize visualzation tools structure in pr #2092

**Pull request merged**